lib/msal-browser/src/interaction_handler/InteractionHandler.ts

```diff
@@ -43,13 +43,23 @@ export abstract class InteractionHandler {
         const cachedNonce = this.browserStorage.getItem(this.browserStorage.generateCacheKey(cachedNonceKey), CacheSchemaType.TEMPORARY) as string;
 
         // Handle code response.
-        const authCode = this.authModule.handleFragmentResponse(locationHash, requestState);
+        const authCodeResponse: AuthorizationCodePayload = this.authModule.handleFragmentResponse(locationHash, requestState);
 
         // Assign code to request
-        this.authCodeRequest.code = authCode;
+        this.authCodeRequest.code = authCodeResponse.code;
+        if (authCodeResponse.cloud_graph_host_name) {
+            const cloudInstanceAuthorityUri = `https://${authCodeResponse.cloud_instance_host_name}/common/`;
+            if (cloudInstanceAuthorityUri !== this.browserStorage.getCachedAuthority()) {
```

Comment:
Will the cached authority be cached with common or the specific tenant? Also same question here about whether the 2 things being compared here are normalized?

Reply:
I believe it is common but I'm not clear on this, need to consult docs for this.

Reply:
I have updated this to be cached with the tenant-id of the tenant used in the authorize call.